model/trigger/task.go

```diff
@@ -214,15 +219,15 @@ func (t *taskTriggers) generateWithAlertRecord(sub *event.Subscription, alertTyp
 }
 
 func (t *taskTriggers) taskOutcome(sub *event.Subscription) (*notification.Notification, error) {
-	if t.data.Status != evergreen.TaskSucceeded && t.data.Status != evergreen.TaskFailed {
+	if t.data.Status != evergreen.TaskSucceeded && !isFailedTaskStatus(t.data.Status) {
 		return nil, nil
 	}
 
 	return t.generate(sub)
 }
 
 func (t *taskTriggers) taskFailure(sub *event.Subscription) (*notification.Notification, error) {
-	if t.data.Status != evergreen.TaskFailed {
+	if !isFailedTaskStatus(t.data.Status) {
```

Comment:
I think you can replace this with a check for t.task.status = failed, since that will include all the states you're checking

Reply:
Have a look at the task status list, it would suggest otherwise.
The following are not things we should consider failure, and I think the reason is obvious: started, unstarted, undispatched, dispatched, inactive.

TaskConflict is something evergreen will resolve on its own.
TaskSystemTimedOut, and TaskSetupFailed appear to be used in TaskEndDetails only.
TaskSystemUnresponsive, TaskSystemTimedOut, TaskTestTimedOut, TaskTimedout, TaskSystemFailed I'm not sure about, and will look into

This leaves TaskFailed, TaskSystemFailed, and TaskTestTimedOut. I will review and get back to you.

Reply:
I haven't read any code, but my intuition is that John may be right. `details.status` is failed, and `details.type` determines what kind of failure it is, e.g., TaskFailed, TaskSystemFailed, etc. I think if you have access to `t.task.status`, you may have enough.

Reply:
The various timed out scenarios will all have status = failed with timeout in the details (but you probably don't care too much about that). https://github.com/evergreen-ci/evergreen/blob/master/model/task/results.go#L15 has the different status scenarios, and I think failed encompasses all the ones you care about

Reply:
Since display tasks will have the result status in the task itself, this approach is the cleanest way to account for both